lib/signer.ex

```diff
@@ -62,7 +62,12 @@ defmodule AntikytheraAws.Signer do
           ad = Time.to_iso_basic(Time.now())
           {ad, Map.put(without_amz_date, "x-amz-date", ad)}
       end
-      sign_ready_headers = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      headers_might_with_st = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      sign_ready_headers    = if service == "s3" and method in [:post, :put] do
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
says that:

> The x-amz-content-sha256 header is required for all AWS Signature Version 4 requests. It provides a hash of the request payload. If there is no payload, you must provide the hash of an empty string.

Is it possible to omit `and method in [:post, :put]` here, so that in case of e.g. `:get` simply `hex_sha256("")` is used?
(I told you about checking HTTP method here but that's inappropriate. Sorry!)

Reply:
Yes, that's right.
fixed in 9dd183d (and include eec2759)